src/main/java/com/patchworkmc/manifest/converter/FieldDescriptorProvider.java

```diff
@@ -0,0 +1,124 @@
+package com.patchworkmc.manifest.converter;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.objectweb.asm.commons.Remapper;
+
+import net.fabricmc.tinyremapper.IMappingProvider;
+
+public class FieldDescriptorProvider {
+	private DescriptorRemapper descriptorRemapper = new DescriptorRemapper();
+	private Map<String, String> descriptors = new HashMap<>();
+
+	public FieldDescriptorProvider(IMappingProvider mappings) {
+		mappings.load(new IMappingProvider.MappingAcceptor() {
+			@Override
+			public void acceptClass(String srcName, String dstName) {
+				descriptorRemapper.acceptClass(srcName, dstName);
+			}
+
+			@Override
+			public void acceptMethod(IMappingProvider.Member method, String dstName) {
+				// NO-OP
+			}
+
+			@Override
+			public void acceptMethodArg(IMappingProvider.Member method, int lvIndex, String dstName) {
+				// NO-OP
+			}
+
+			@Override
+			public void acceptMethodVar(IMappingProvider.Member method, int lvIndex, int startOpIdx, int asmIndex, String dstName) {
+				// NO-OP
+			}
+
+			@Override
+			public void acceptField(IMappingProvider.Member field, String dstName) {
+				// NO-OP
+			}
+		});
+
+		mappings.load(new IMappingProvider.MappingAcceptor() {
```

Comment:
I don't like how we have to load the mappings twice here but it's not worth reevaluating in this PR.

Reply:
The problem is any descriptor could arbitrary reference any class in any order. I'd like to find another way as well.